staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -33,21 +32,17 @@ var logRegistry = NewLogFormatRegistry()
 // LogFormatRegistry store klog format registry
 type LogFormatRegistry struct {
 	registry map[string]logr.Logger
-	mu       sync.Mutex
 }
 
 // NewLogFormatRegistry return new init LogFormatRegistry struct
 func NewLogFormatRegistry() *LogFormatRegistry {
 	return &LogFormatRegistry{
 		registry: make(map[string]logr.Logger),
-		mu:       sync.Mutex{},
 	}
 }
 
 // Register new log format registry to global logRegistry
 func (lfr *LogFormatRegistry) Register(name string, logger logr.Logger) error {
-	lfr.mu.Lock()
-	defer lfr.mu.Unlock()
 	if _, ok := lfr.registry[name]; ok {
```

Comment:
I'm all for the locks being removed. However I believe part of the reasoning for removing the locks was that you can't reasonably add new loggers after initialization (or more specifically after validation). I think the API should reflect that. My preference would be to make Set/Register private. I know that lavalamp mentioned making them blow up if called after validation. (I would be fine with that as well)

Reply:
Yeah it seems unlikely that people will accidentally make initialization multi-threaded.

Accidentally registering post-validate / flag generation seems like a more likely error, I think it'd be reasonable to set a boolean and panic in that condition.